client/app/pages/dashboards/widget.js

```diff
@@ -23,18 +23,25 @@ const EditTextBoxComponent = {
         this.saveInProgress = false;
       });
     };
+    this.closeWithoutSave = () => {
+      this.widget.text = this.widget.existing_text;
+      this.close();
+    };
   },
 };
 
 function DashboardWidgetCtrl($location, $uibModal, $window, Events, currentUser) {
   this.canViewQuery = currentUser.hasPermission('view_query');
 
   this.editTextBox = () => {
+    this.widget.existing_text = this.widget.text;
```

Comment:
I think it will be better to keep this logic in the component. So the component will make a copy of the original value and restore to it when closed instead of saved.

Another option is to use a different object as the `ng-model` of the `<textarea>` and only on save assign it to the widget. This way no extra logic is needed on close and you can bring back keyboard support.